docs/book/src/cronjob-tutorial/testdata/project/api/v1/cronjob_webhook.go

```diff
@@ -59,31 +69,63 @@ The meaning of each marker can be found [here](/reference/markers/webhook.md).
 // +kubebuilder:webhook:path=/mutate-batch-tutorial-kubebuilder-io-v1-cronjob,mutating=true,failurePolicy=fail,groups=batch.tutorial.kubebuilder.io,resources=cronjobs,verbs=create;update,versions=v1,name=mcronjob.kb.io,sideEffects=None,admissionReviewVersions=v1
 
 /*
-We use the `webhook.Defaulter` interface to set defaults to our CRD.
+We use the `webhook.CustomDefaulter` interface to set defaults to our CRD.
 A webhook will automatically be served that calls this defaulting.
 
 The `Default` method is expected to mutate the receiver, setting the defaults.
 */
 
-var _ webhook.Defaulter = &CronJob{}
+// +kubebuilder:object:generate=false
+// CronJobCustomDefaulter struct is responsible for setting default values on the custom resource of the
+// Kind CronJob when those are created or updated.
+//
+// NOTE: The +kubebuilder:object:generate=false marker prevents controller-gen from generating DeepCopy methods,
+// as it is used only for temporary operations and does not need to be deeply copied.
+type CronJobCustomDefaulter struct {
+	cronjob *CronJob
```

Comment:
This is not concurrency safe. What if the webhook is called for two CronJobs at the same time?

I would just drop the field and pass the cronJob into the setDefault func

Reply:
That is a very good catcher!!! 
I will change.